packages/transactions/src/decompile-transaction.ts

```diff
@@ -58,6 +59,54 @@ function getAccountMetas(message: CompiledMessage): IAccountMeta[] {
     return accountMetas;
 }
 
+export type LookupTables = { [lookupTableAddress: Address]: Address[] };
+
+function getAddressLookupMetas(
+    transactionLookups: ReturnType<typeof getCompiledAddressTableLookups>,
+    knownLookups: LookupTables,
+): IAccountLookupMeta[] {
+    // check that all message lookups are known
+    const transactionLookupAddresses = transactionLookups.map(l => l.lookupTableAddress);
+    const missing = transactionLookupAddresses.filter(a => knownLookups[a] === undefined);
+    if (missing.length > 0) {
+        const missingAddresses = missing.join(', ');
+        // TODO: coded error.
+        throw new Error(`Transaction includes missing address lookup tables: [${missingAddresses}]`);
+    }
```

Comment:
I stared at this for a while, thinking about eliminating the additional iteration(s). However, I actually like how you've opted for a better error here, one that describes exactly which tables are missing. Nice.